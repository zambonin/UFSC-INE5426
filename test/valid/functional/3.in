int t[10], output[10]
output = map(lambda int x -> x + 2, t)
