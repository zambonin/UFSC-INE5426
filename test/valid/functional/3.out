int array: t (size: 10), output (size: 10)
int array fun: t_map (params: int array t)
  int fun: lambda (params: int x)
    ret + x 2
  int var: t_ti
  int array: t_ta (size: 10)
  for: = t_ti 0, < t_ti [len] t, = t_ti + t_ti 1
  do:
    = [index] t_ta t_ti lambda[1 params] [index] t t_ti
  ret t_ta
= output t_map[1 params] t
